modules/core/src/main/java/org/apache/ignite/internal/binary/BinaryContext.java

```diff
@@ -706,6 +722,39 @@ public BinaryClassDescriptor descriptorForTypeId(
         return desc;
     }
 
+    /**
+     * Creates descriptor without registration.
+     *
+     * @param cls Class.
+     * @return Binary class descriptor.
+     */
+    @NotNull private BinaryClassDescriptor createNoneCacheClassDescriptor(Class cls) {
+        String clsName = cls.getName();
+
+        BinaryInternalMapper mapper = userTypeMapper(clsName);
+
+        int typeId = mapper.typeId(clsName);
+
+        String typeName = mapper.typeName(clsName);
+
+        BinarySerializer serializer = serializerForClass(cls);
+
+        String affFieldName = affinityFieldName(cls);
+
+        return new BinaryClassDescriptor(this,
+            cls,
+            true,
+            typeId,
+            typeName,
+            affFieldName,
+            mapper,
+            serializer,
+            true,
+            true,
+            false
```

Comment:
I am not sure, but this parameter (exclude == false) will be able to give a side-efect in some cases, where we use excluded types.

Reply:
Shouldn't be a problem.
If instance is already present as a binary object, its class is not included.